blocksync/metrics.go

```diff
@@ -16,4 +22,44 @@ const (
 type Metrics struct {
 	// Whether or not a node is block syncing. 1 if yes, 0 if no.
 	Syncing metrics.Gauge
+	// Number of transactions in the latest block.
+	NumTxs metrics.Gauge
+	// Total number of transactions.
+	TotalTxs metrics.Gauge
+	// Size of the latest block.
+	BlockSizeBytes metrics.Gauge
+	// The height of the latest block.
+	LatestBlockHeight metrics.Gauge
+}
+
+func (m *Metrics) recordBlockMetrics(block *types.Block) {
+	m.NumTxs.Set(float64(len(block.Data.Txs)))
+	m.TotalTxs.Add(float64(len(block.Data.Txs)))
+	m.BlockSizeBytes.Set(float64(block.Size()))
+	m.LatestBlockHeight.Set(float64(block.Height))
+}
+
+// unregisterGauge removes a gauge metric from the Prometheus registry. No new
+// data will be generated in the http /metrics endpoint for this metric.
+func unregisterGauge(g metrics.Gauge) bool {
```

Comment:
I discussed this with @hvanz on Slack - I really don't think we should be unregistering Prometheus metrics. It seems risky, especially if external metrics systems come to rely on specific metrics being present. Plus it's really non-standard.

If we do really want to unregister metrics, there needs to be a _very_ good reason.

Reply:
We discussed it synchronously. The best solution will be to unify certain metrics, such as the height, into a node-wide set of metrics. But that will need a bit more work on infra.
I agree we probably want to not unregister upon catching up, even if the resulting graph looks somewhat awkward.

Reply:
Thanks for the discussions! As a result, we won't unregister the metrics, and we'll have to think about a better solution for metrics at the node level.